syntax = "proto3";

enum RequestType {
    PING = 1;
    SET_PIN_DIRECTION = 2;
    SET_PIN_VALUE = 3;
    GET_PIN_DIRECTION = 4;
    GET_PIN_VALUE = 5;
}

enum PinValue {
    LOW = 1;
    HIGH = 2;
    INPUT = 3;
    OUTPUT = 4;
}

message PicohaDioRequest {
    RequestType type = 1;
    uint32 pin_num = 2;
    PinValue value = 3;
}

enum AnswerType {
    SUCCESS = 1;
    FAILURE = 2;
}

message PicohaDioAnswer {
    AnswerType type = 1;
    optional PinValue value = 2;
    optional String error_message = 3;
}
